src/main/java/spoon/support/reflect/code/CtLambdaImpl.java

```diff
@@ -101,7 +105,32 @@ public String getSimpleName() {
 			//it can be null in noclasspath mode, so we do not know which method is called, by lambda
 			return null;
 		}
-		CtType<T> lambdaType = lambdaTypeRef.getTypeDeclaration();
+		CtMethod<?> lambdaExecutableMethod = null;
+		if (!(lambdaTypeRef instanceof CtIntersectionTypeReference)) {
+			return getOverriddenAux(lambdaTypeRef);
+		} else {
+			CtElement parent = lambdaTypeRef.getParent();
+			CtTypeReference<?> parentTypeReference = null;
+			if (parent != null && parent instanceof CtLocalVariable) {
+				parentTypeReference = ((CtLocalVariable) parent).getType();
+			} else if (parent != null && parent instanceof CtAssignment) {
+				parentTypeReference = ((CtAssignment) parent).getAssigned().getType();
+			}
+			for (CtTypeReference<?> ctTypeReference : ((CtIntersectionTypeReference<?>) lambdaTypeRef).getBounds()) {
+				CtMethod<?> tmp = getOverriddenAux(ctTypeReference);
+				if (tmp != null && (lambdaExecutableMethod == null || ctTypeReference.equals(parentTypeReference))) {
+					lambdaExecutableMethod = tmp;
+				}
+			}
+		}
+		if (lambdaExecutableMethod == null) {
```

Comment:
do we need to move this check here? could we keep it at the previous place?

Reply:
It follows the standard compiler behavior.
That is, at least one interface should have one method,
thus, some of these interfaces might not even declare any methods.

But as a side note, I just match the type that will be taken by the lambda in its parent,
but I did not implement a real type intersection, thus I am less strict than the compiler.
A great improvement would be to implement this type intersection in spoon.
But it is too complicated for me :/ and I would not be useful for my use case.

Reply:
OK for me, thanks.